### PR TITLE
Basic tests

### PR DESCRIPTION
**Pull request by gegorov**
In this PR I added basic tests.

code coverage is 100% :)
image

### CODE REVIEW

File: src/app/courses-list/courses-list.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,72 @@
...
-  });
-
-  it("should create", () => {
-    expect(component).toBeTruthy();
-  });
+    let component: CoursesListComponent;
+    let fixture: ComponentFixture<CoursesListComponent>;
+
+    let courseService: CourseService;
+    const CourseServiceStub: Partial<CourseService> = {
```
Comment (kirillpanov) [inline src/app/courses-list/courses-list.component.spec.ts]: Take a look please at jasmine.createSpyObj

File: src/app/courses-list/courses-list.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,72 @@
...
+        const data = 123;
+
+        component.onNotify(data);
+        fixture.detectChanges();
+        expect(console.log).toHaveBeenCalledWith("Delete movie with ID# ", data);
+    });
+
+    it("should call courseService in ngOnInit", () => {
+        const getCoursesSpy = spyOn(courseService, "getCourses");
+        component.ngOnInit();
```
Comment (kirillpanov) [inline src/app/courses-list/courses-list.component.spec.ts]: it is better to call fixture.detectChanges()

File: src/app/core/services/course-service/course.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { CourseService } from "./course.service";
+import { COURSES } from "../../helpers/mock-courses";
+import { ICourse } from "../../models/course.interface";
+
+describe("CourseService", () => {
+    const courseService = new CourseService();
```
Comment (kirillpanov) [inline src/app/core/services/course-service/course.service.spec.ts]: Please, use TestBed

File: src/app/app.component.spec.ts [hunk 4]
```diff
@@ -26,10 +28,11 @@ describe("AppComponent", () => {
     expect(app.title).toEqual("angular-gmp");
   });
 
-  it("should render title", () => {
+  it("should render course items", () => {
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/app.component.spec.ts]: removed

File: src/app/core/services/course-service/course.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+describe("CourseService", () => {
+    const courseService = new CourseService();
+    const mockedCourses: Array<ICourse> = COURSES;
+
+    it("should return an observable", (done) => {
+        const courses$ = courseService.getCourses();
+        courses$.pipe().subscribe(
+            (courses: Array<ICourse>) => {
+                expect(courses).toEqual(mockedCourses);
+                done();
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/core/services/course-service/course.service.spec.ts]: ok. noted!

File: src/app/courses-list/course/course.component.spec.ts [hunk 6]
```diff
@@ -15,10 +15,30 @@ describe("CourseComponent", () => {
...
+    it("should emit correct id when onDelete called", () => {
+        spyOn(component.removeItem, "emit");
+
+        const nativeElement = fixture.nativeElement;
+        const button = nativeElement.querySelector(".btn__actions-delete");
+        button.dispatchEvent(new Event("click"));
+
+        fixture.detectChanges();
+        expect(component.removeItem.emit).toHaveBeenCalledWith(component.course.id);
+
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/courses-list/course/course.component.spec.ts]: done

File: src/app/courses-list/course/course.component.spec.ts [hunk 7]
```diff
@@ -15,10 +15,30 @@ describe("CourseComponent", () => {
...
     it("should create", () => {
         expect(component).toBeTruthy();
     });
+
+    it("should emit correct id when onDelete called", () => {
+        spyOn(component.removeItem, "emit");
+
+        const nativeElement = fixture.nativeElement;
+        const button = nativeElement.querySelector(".btn__actions-delete");
+        button.dispatchEvent(new Event("click"));
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/courses-list/course/course.component.spec.ts]: done

File: src/app/courses-list/courses-list.component.spec.ts [hunk 8]
```diff
@@ -1,25 +1,72 @@
 import { async, ComponentFixture, TestBed } from "@angular/core/testing";
 
 import { CoursesListComponent } from "./courses-list.component";
+import { CourseComponent } from "./course/course.component";
+import { SearchComponent } from "./search/search.component";
+import { FormsModule } from "@angular/forms";
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/courses-list/courses-list.component.spec.ts]: fixed

File: src/app/courses-list/courses-list.component.spec.ts [hunk 9]
```diff
@@ -1,25 +1,72 @@
...
+import { ServiceModule } from "../core/services/index";
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/courses-list/courses-list.component.spec.ts]: fixed

File: src/app/courses-list/courses-list.component.spec.ts [hunk 10]
```diff
@@ -1,25 +1,72 @@
...
+
+import { CourseService} from "../core/services/index";
```
[the remark this answers is not in the document]
Comment (gegorov) [reply src/app/courses-list/courses-list.component.spec.ts]: fixed

**Pull request merged**